code/datums/antagonists/datum_changeling.dm

```diff
@@ -0,0 +1,60 @@
+/datum/antagonist/changeling
+	name = ROLE_CHANGELING
+
+	text_on_lose = "You are no longer a changeling!"
+
+	give_special_equipment = FALSE
+
+	number_of_possible_objectives = 2
+	possible_objectives = list(/datum/objective/steal, /datum/objective/assassinate, /datum/objective/maroon)
+
+/datum/antagonist/changeling/apply_innate_effects()
+	if(!owner)
+		return
+	if(owner.current)
+		owner.current.make_changeling()
+
+	ticker.mode.update_changeling_icons_added(owner)
```

Comment:
Should probably move the antag hud adding and removing code into the threat datum.
Rather than having a separate proc defined for each antag type, it could be generalized to simply use arguments as the key to add to the right HUD.